edk2toolext/tests/versioninfo_tool_test.py

```diff
@@ -0,0 +1,217 @@
+
+
+class VersionInfoTest(unittest.TestCase):
+    # @classmethod
+    # def setUpClass(cls):
+    #     cls.temp_dir = tempfile.mkdtemp()
+    #     cls.dummy_json = os.path.join(cls.temp_dir, DUMMY_JSON_FILE_NAME + '.json.orig')
+    #     cls.dummy_exe_src = os.path.join(cls.temp_dir, DUMMY_EXE_SRC_NAME + '.cpp')
+    #     cls.dummy_exe_makefile = os.path.join(cls.temp_dir, DUMMY_EXE_MAKEFILE_NAME)
+
+    #     with open(cls.dummy_json, 'w') as dummy_file:
+    #         json.dump(DUMMY_VALID_JSON, dummy_file)
+    #     with open(cls.dummy_exe_src, 'w') as dummy_src:
+    #         dummy_src.write(DUMMY_EXE_SOURCE)
+    #     with open(cls.dummy_exe_makefile, 'w') as dummy_makefile:
+    #         dummy_makefile.write(DUMMY_EXE_MAKEFILE)
+
+    @unittest.skipUnless(sys.platform.startswith("win"), "requires Windows")
+    def test_encode_decode_windows(self):
+        # Find VS build tools
+        (rc, vc_install_path) = locate_tools.FindWithVsWhere(vs_version="vs2019")
+        if rc != 0 or vc_install_path is None or not os.path.exists(vc_install_path):
+            logging.fatal("Cannot locate VS build tools install path")
+            self.fail()
+
+        vc_ver_path = os.path.join(vc_install_path, "VC", "Tools", "MSVC")
+        if not os.path.isdir(vc_ver_path):
+            logging.fatal("Cannot locate VS build tools directory")
+            self.fail()
+
+        vc_ver = os.listdir(vc_ver_path)[-1].strip()
+        if vc_ver is None:
+            logging.fatal("Cannot locate VS build tools version")
+            self.fail()
+
+        # Set VS env variables
+        vs2019_prefix = os.path.join(vc_install_path, "VC", "Tools", "MSVC", vc_ver)
+        vs2019_prefix += os.path.sep
+        shell_environment.GetEnvironment().set_shell_var("VS2019_PREFIX", vs2019_prefix)
+        shell_environment.GetEnvironment().set_shell_var("VS2019_HOST", "x64")
+        shell_env = shell_environment.GetEnvironment()
+        vs_vars = locate_tools.QueryVcVariables(VS2019_INTERESTING_KEYS, "x64", vs_version="vs2019")
+        for (k, v) in vs_vars.items():
+            shell_env.set_shell_var(k, v)
+
+        temp_dir = tempfile.mkdtemp()
+        dummy_json = os.path.join(temp_dir, DUMMY_JSON_FILE_NAME + '.json.orig')
+        dummy_exe_src = os.path.join(temp_dir, DUMMY_EXE_SRC_NAME + '.cpp')
+        dummy_exe_makefile = os.path.join(temp_dir, DUMMY_EXE_MAKEFILE_NAME)
+
+        with open(dummy_json, 'w') as dummy_file:
+            json.dump(DUMMY_VALID_JSON, dummy_file)
+        with open(dummy_exe_src, 'w') as dummy_src:
+            dummy_src.write(DUMMY_EXE_SOURCE)
+        with open(dummy_exe_makefile, 'w') as dummy_makefile:
+            dummy_makefile.write(DUMMY_EXE_MAKEFILE)
+
+        cli_params = [dummy_json, temp_dir]
+        parsed_args = versioninfo_tool.get_cli_options(cli_params)
+        versioninfo_tool.service_request(parsed_args)
+        cd = os.getcwd()
+        os.system('cd %s && nmake rsrc && cd %s' % (temp_dir, cd))
```

Comment:
Probably better to use RunCmd here. You can pass in the directory you want it to execute in